source-code/sdk-js/src/config/index.ts

```diff
@@ -0,0 +1,47 @@
+import {
+	Config as InlangConfig,
+	EnvironmentFunctions,
+	initialize$import,
+} from "@inlang/core/config"
+
+// this is a copy from `source-code/website/src/pages/editor/@host/@owner/@repository/State.tsx`
+// TODO: move functionality into `core/config`
+
+async function readInlangConfig(args: {
+	fs: typeof import("memfs").fs
+}): Promise<InlangConfig | undefined> {
+	try {
+		const environmentFunctions: EnvironmentFunctions = {
+			$import: initialize$import({
+				workingDirectory: "/",
+				fs: args.fs.promises,
+				fetch,
+			}),
+			$fs: args.fs.promises,
+		}
+		const file = await args.fs.promises.readFile("./inlang.config.js", "utf-8")
+		const withMimeType =
+			"data:application/javascript;base64," + Buffer.from(file).toString("base64")
+
+		const module = await import(/* @vite-ignore */ withMimeType)
+		const config: InlangConfig = await module.defineConfig({
+			...environmentFunctions,
+		})
+
+		return config
+	} catch (error) {
+		// TODO: don't throw errors
```

Comment:
how do we wan't to deal with exceptions? Throwing errors is bad! The developer who uses this function needs to know implementation details or else he will not know that something can go wrong here.
We should return exceptions instead. 
https://github.com/ivanhofer/exceptionally or `[data, exception]` tuple?

Reply:
SolidJS requires thrown Errors for ErrorBoundaries. 

As stated in https://github.com/inlang/inlang/pull/473/files/f1068c12d11fc3175a5e171fce9af702c598564e#r1150146838, you don't need this function.

Reply:
but I still need an error handling. Reading from fs can also go wrong

Reply:
Not if you don't have a `readInlangConfig` function^^. If you decide to use one, just return whatever you'd like. The decision to use a result here or not is not exposed to the outside world.

Reply:
I know, but internally we should also use the same thing everywhere and not have 5 different solutions

Reply:
> I know, but internally we should also use the same thing everywhere and not have 5 different solutions 

I agree. If we want the same type everywhere, I would either use the existing `Result` type (but it's too complicated) or use a simple union type with no extra bells and whistles

Reply:
I will just return `undefined` for now unti lwe have an official solution